### PR TITLE
Fixes #14586: Update relay component

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/14586

### CODE REVIEW

**Comment by amousset** [comment]
Commit modified

File: relay/sources/relayd/src/data/reporting.rs [hunk 1]
```diff
@@ -129,40 +126,48 @@ named!(parse_runlog<CompleteStr, Vec<RawReport>>,
...
+    // no line break inside a filed (except message)
+    // handle partial reports without breaking following ones
     logs: log_entries >>
     rudder_report_begin >>
     policy: take_until_and_consume_s!("@@") >>
     event_type: take_until_and_consume_s!("@@") >>
     rule_id: take_until_and_consume_s!("@@") >>
     directive_id: take_until_and_consume_s!("@@") >>
-    serial: take_until_and_consume_s!("@@") >>
+    serial: map_res!(take_until_and_consume_s!("@@"), parse_i32) >>
```
Comment (ncharles) [inline relay/sources/relayd/src/data/reporting.rs]: serial should always be 0

Comment (ncharles) [reply relay/sources/relayd/src/data/reporting.rs]: meaning: there is no way the serial can be set to something else than 0 in Rudder 4.3 and more

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

File: relay/sources/relayd/src/data/reporting.rs (hunk 1 above)
Comment (amousset) [reply relay/sources/relayd/src/data/reporting.rs]: The idea is to allow reusing this field if we want it in the future, without having to upgrade all relays.

Comment (amousset) [reply relay/sources/relayd/src/data/reporting.rs]: (specially as it is cheap and just as simple to implement)

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
PR rebased

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
OK, squash merging this PR

**Comment by amousset** [comment]
OK, squash merging this PR

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Pull request merged**